src/nvim/regexp_nfa.c

```diff
@@ -6075,6 +6079,13 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
     if (got_int) {
```

Comment:
Oh, not a race, just a missing init in the unit tests. Diagnosed by @oni-link at https://github.com/neovim/neovim/pull/4070#discussion_r50626558